### ISSUE TITLE
Make get() return a stream of Result

### ISSUE THREAD

**Issue by Mallets** [eclipse-zenoh/zenoh#242]
When issuing a query via a `get()` operation, a stream of `Sample` is received.
However, while receiving all the `Sample`, some error may occur while resolving the query.

E.g., the query may have timed-out, the target queryable(s) didn't understand the query selectors, consolidation can not be fully performed for memory consumption reasons (e.g., too many resource keys), etc.

As a consequence of that, it is advisable to return a stream of `Result` so as to inform the user wether any error occurred.

**Comment by Mallets** [comment]
99835bbabe100656e0824c06806765d01003b6bb has been merged into api-changes branch.
This issue will need to be closed manually once api-changes is merged into master.

**Comment by Mallets** [comment]
Merged into master.
